File: src/main/java/nobs/library/LibraryRepository.java
package nobs.library;

import nobs.book.BookRepository;

public class LibraryRepository {
    private BookRepository bookRepository;

    public LibraryRepository(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    public Library getLibrary() {
        return new Library(bookRepository.getBooks());
    }
}

File: src/main/webapp/modules/library/libraryRepository.js
define(function() {

    var LibraryRepository = function() {
        var urlRoot = "http://localhost:8080/rest/library/";

        var getLibrary = function(libraryId) {
            var libraryPromise = $.ajax(urlRoot + libraryId, {
                accepts: {
                    json: 'application/json'
                }
            });

            return libraryPromise;
        };

        return {
            getLibrary: getLibrary
        }
    }

    return LibraryRepository;
});